awx/ui_next/src/screens/Setting/Logging/LoggingEdit/LoggingEdit.jsx

```diff
@@ -1,23 +1,272 @@
+      {isLoading && <ContentLoading />}
+      {!isLoading && error && <ContentError error={error} />}
+      {!isLoading && logging && (
+        <Formik
+          initialValues={{ ...initialValues(logging) }}
+          onSubmit={handleSubmit}
         >
-          {i18n._(t`Cancel`)}
-        </Button>
-      </CardActionsRow>
+          {formik => {
+            return (
+              <Form autoComplete="off" onSubmit={formik.handleSubmit}>
+                <FormColumnLayout>
+                  <BooleanField
+                    name="LOG_AGGREGATOR_ENABLED"
+                    config={logging.LOG_AGGREGATOR_ENABLED}
+                    ariaLabel={i18n._(t`Enable external logging`)}
+                    disabled={
+                      !formik.values.LOG_AGGREGATOR_ENABLED &&
+                      (!formik.values.LOG_AGGREGATOR_HOST ||
+                        !formik.values.LOG_AGGREGATOR_TYPE)
+                    }
+                  />
+                  <InputField
+                    name="LOG_AGGREGATOR_HOST"
+                    config={logging.LOG_AGGREGATOR_HOST}
+                    isRequired={Boolean(formik.values.LOG_AGGREGATOR_ENABLED)}
+                  />
+                  <InputField
+                    name="LOG_AGGREGATOR_PORT"
+                    config={logging.LOG_AGGREGATOR_PORT}
+                    type="number"
+                  />
+                  <ChoiceField
+                    name="LOG_AGGREGATOR_TYPE"
+                    config={logging.LOG_AGGREGATOR_TYPE}
+                    isRequired={Boolean(formik.values.LOG_AGGREGATOR_ENABLED)}
+                  />
+                  <InputField
+                    name="LOG_AGGREGATOR_USERNAME"
+                    config={logging.LOG_AGGREGATOR_USERNAME}
+                  />
+                  <EncryptedField
+                    name="LOG_AGGREGATOR_PASSWORD"
+                    config={logging.LOG_AGGREGATOR_PASSWORD}
+                  />
+                  <BooleanField
+                    name="LOG_AGGREGATOR_INDIVIDUAL_FACTS"
+                    ariaLabel={i18n._(
+                      t`Enable log system tracking facts individually`
+                    )}
+                    config={logging.LOG_AGGREGATOR_INDIVIDUAL_FACTS}
+                  />
+                  <ChoiceField
+                    name="LOG_AGGREGATOR_PROTOCOL"
+                    config={logging.LOG_AGGREGATOR_PROTOCOL}
+                  />
+                  <ChoiceField
+                    name="LOG_AGGREGATOR_LEVEL"
+                    config={logging.LOG_AGGREGATOR_LEVEL}
+                  />
+                  {['tcp', 'https'].includes(
+                    formik.values.LOG_AGGREGATOR_PROTOCOL
+                  ) && (
+                    <InputField
+                      name="LOG_AGGREGATOR_TCP_TIMEOUT"
+                      config={logging.LOG_AGGREGATOR_TCP_TIMEOUT}
+                      type="number"
+                      isRequired
```

Comment:
What about we add more specific validator for `type="number"` ? We do have `number` or `integer` validators.

![image](https://user-images.githubusercontent.com/9053044/97589995-8e887200-19d4-11eb-8f5a-3e189292df35.png)

Reply:
Of course Firefox will let us type non-numeric characters in a `type="number"` input field. See:https://bugzilla.mozilla.org/show_bug.cgi?id=1398528
If I remove the type attribute from TextInput, the validators will work as expected.